app/verify_email/_client.tsx

```diff
@@ -0,0 +1,106 @@
+"use client";
+
+import { Button } from "@headlessui/react";
+import { AlertCircle, CheckCircle, Loader } from "lucide-react";
+import { useRouter, useSearchParams } from "next/navigation";
+import React, { useEffect, useState } from "react";
+
+function Content({ userId }: { userId: string }) {
+  const params = useSearchParams();
+  const router = useRouter();
+  const [status, setStatus] = useState<
+    "idle" | "loading" | "success" | "error"
+  >("idle");
+  const [message, setMessage] = useState("");
+  const [token, setToken] = useState<string | null>(null);
+
+  useEffect(() => {
+    const tokenParam = params.get("token");
+    if (tokenParam && !token) {
+      setToken(tokenParam);
+    }
+  }, [params, token]);
+
+  useEffect(() => {
+    const verifyEmail = async () => {
+      if (!token) {
+        setStatus("error");
+        setMessage(
+          "No verification token found. Please check your email for the correct link.",
+        );
+        return;
+      }
+      setStatus("loading");
+
+      try {
+        const res = await fetch(`/api/verify_email?token=${token}`, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ userId }),
+        });
```

Comment:
The user should be able to click a link to a GET route which will auto-magically validate the token (instead of having to post a request). This ensures that the user clicked their email always.